setup.sh

```diff
@@ -54,13 +54,15 @@ find . -regex "^.+.\(sh\|py\)" | xargs chmod a+x
 
 echo "RSYNC'ing /root/spark-ec2 to other cluster nodes..."
 rsync_start_time="$(date +'%s')"
-for node in $SLAVES $OTHER_MASTERS; do
-  echo $node
-  rsync -e "ssh $SSH_OPTS" -az /root/spark-ec2 $node:/root &
-  scp $SSH_OPTS ~/.ssh/id_rsa $node:.ssh &
-  sleep 0.1
-done
-wait
+parallel --quote rsync -e "ssh $SSH_OPTS" -az /root/spark-ec2 {}:/root ::: $SLAVES $OTHER_MASTERS
```

Comment:
@nchammas - Can we postpone this change to use `parallel` to a later PR i.e. once we have the Packer built AMIs ? I ask this because if we merge this PR right now this might not be backwards compatible with existing AMIs